tests/usage.test.ts

```diff
@@ -1,81 +1,90 @@
+
+/**
+ * Helper function to walk /examples directory and collect generated JSON files
+ */
+const getJSONExamples = <T>(schemaPath: string): T[] => {
+  const examples: T[] = [];
+
+  const walkDirectory = (dir: string) => {
+    const files = fs.readdirSync(dir);
+
+    for (const file of files) {
+      const filePath = path.join(dir, file);
+
+      if (fs.statSync(filePath).isDirectory()) {
+        walkDirectory(filePath);
+      } else if (path.extname(filePath) === '.json') {
+        const example = JSON.parse(fs.readFileSync(filePath, 'utf8'));
+        examples.push(example);
+      }
+    }
+  };
+
+  const examplesDir = path.join(__dirname, `../examples/${schemaPath}`);
+  walkDirectory(examplesDir);
+
+  return examples;
+};
+
+const schemas = [
+  {
+    name: 'DigitalPlanningApplication',
+    schema: digitalPlanningApplicationSchema,
+    examples: getJSONExamples<DigitalPlanningApplication>(
+      'digitalPlanningApplication'
+    ),
+  },
 ];
 
-describe("parsing using the 'jsonschema' library", () => {
-  examplesToTest.forEach(example => {
-    test(`accepts a valid example: ${example.data.application.type.description}`, () => {
-      const validator = new Validator();
-      const result = validator.validate(example, generatedSchema, {
-        disableFormat: true,
+describe.each(schemas)('$name', ({schema, examples}) => {
+  const validator = new Validator();
+
+  describe("parsing using the 'jsonschema' library", () => {
+    describe.each(examples)('$data.application.type.description', example => {
+      test('accepts a valid example', async () => {
+        const result = validator.validate(example, schema as Schema);
+
+        expect(result.errors).toHaveLength(0);
       });
 
-      expect(result.errors).toHaveLength(0);
-    });
-  });
+      test('rejects an invalid example', () => {
+        const invalidExample = {foo: 'bar'};
+        const result = validator.validate(invalidExample, schema as Schema);
 
-  test('rejects an invalid example', () => {
-    const validator = new Validator();
-    const invalidExample = {foo: 'bar'};
-    const result = validator.validate(invalidExample, generatedSchema, {
-      disableFormat: true,
+        expect(result.errors).not.toHaveLength(0);
+      });
     });
-
-    expect(result.errors).not.toHaveLength(0);
   });
-});
 
-describe("parsing using the 'ajv' library", () => {
-  examplesToTest.forEach(example => {
-    test(`accepts a valid example: ${example.data.application.type.description}`, () => {
-      // addFormats() is required for types UUID, email, datetime etc
-      const ajv = addFormats(new Ajv({allowUnionTypes: true}));
-      const validate = ajv.compile(generatedSchema);
-      const isValid = validate(example);
+  describe("parsing using the 'ajv' library", () => {
+    // addFormats() is required for types UUID, email, datetime etc
+    const ajv = addFormats(new Ajv({allowUnionTypes: true}));
+    const validate = ajv.compile(schema);
```

Comment:
Moving the instantiation of `ajv` and `jsonschema` up to the `describe()` block and not repeating in it each test speeds tests up from ~55s to ~3s.

Reply:
Huge ! Thanks for spotting that misconfiguration